examples/qgc_proxy_px4/qgc_proxy_px4.cpp

```diff
@@ -0,0 +1,183 @@
+#include <mavsdk/mavsdk.h>
+#include <mavsdk/plugins/mavlink_passthrough/mavlink_passthrough.h>
+#include <iostream>
+#include <thread>
+#include <queue>
+#include <mutex>
+#include <condition_variable>
+#include <atomic>
+#include <csignal>
+#include <map>
+#include <optional>
+
+using namespace mavsdk;
+
+// Constants
+constexpr size_t MAX_QUEUE_SIZE = 1000;
+constexpr int PROXY_SYSID_PX4 = 127;
+constexpr int PROXY_SYSID_QGC = 128;
+constexpr int TIMEOUT_SECONDS = 10;
+
+#define PX4_NAME "PX4"
+#define QGC_NAME "QGC"
+
+// Global variables
+std::atomic<bool> running{true};
+std::map<std::string, std::queue<mavlink_message_t>> message_queues;
+std::map<std::string, std::unique_ptr<std::mutex>> queue_mutexes;
+std::map<std::string, std::unique_ptr<std::condition_variable>> queue_cvs;
+std::map<std::string, std::shared_ptr<MavlinkPassthrough>> passthrough_map;
+std::mutex passthrough_map_mutex;
+
+// Thread-safe logging
+void log_message(const std::string& level, const std::string& message) {
+    static std::mutex log_mutex;
+    std::lock_guard<std::mutex> lock(log_mutex);
+    std::cout << "[" << level << "] " << message << std::endl;
+}
```

Comment:
Maybe we should actually do this within MAVSDK, right?

Reply:
How can to do this with MAVSDK ? Is there a function to this ?